src/main/kotlin/com/openlattice/shuttle/IntegrationService.kt

```diff
@@ -0,0 +1,279 @@
+package com.openlattice.shuttle
+
+import com.dataloom.mappers.ObjectMappers
+import com.google.common.base.Preconditions.checkState
+import com.google.common.util.concurrent.MoreExecutors
+import com.hazelcast.core.HazelcastInstance
+import com.hazelcast.query.Predicates
+import com.openlattice.authorization.HazelcastAclKeyReservationService
+import com.openlattice.authorization.Principals
+import com.openlattice.client.ApiClient
+import com.openlattice.data.DataIntegrationApi
+import com.openlattice.data.EntityKeyIdService
+import com.openlattice.data.S3Api
+import com.openlattice.data.integration.IntegrationDestination
+import com.openlattice.data.integration.StorageDestination
+import com.openlattice.data.integration.destinations.PostgresDestination
+import com.openlattice.data.integration.destinations.PostgresS3Destination
+import com.openlattice.datastore.services.EdmManager
+import com.openlattice.datastore.services.EntitySetManager
+import com.openlattice.edm.EntitySet
+import com.openlattice.edm.type.EntityType
+import com.openlattice.edm.type.PropertyType
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.hazelcast.HazelcastQueue
+import com.openlattice.retrofit.RhizomeByteConverterFactory
+import com.openlattice.retrofit.RhizomeCallAdapterFactory
+import com.openlattice.retrofit.RhizomeJacksonConverterFactory
+import com.openlattice.rhizome.proxy.RetrofitBuilders
+import com.openlattice.shuttle.control.*
+import com.openlattice.shuttle.hazelcast.processors.UpdateIntegrationEntryProcessor
+import com.openlattice.shuttle.logs.Blackbox
+import com.openlattice.shuttle.mapstore.INTEGRATION_STATUS
+import com.openlattice.shuttle.payload.JdbcPayload
+import com.openlattice.shuttle.payload.Payload
+import com.zaxxer.hikari.HikariConfig
+import com.zaxxer.hikari.HikariDataSource
+import org.apache.olingo.commons.api.edm.FullQualifiedName
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import retrofit2.Retrofit
+import java.net.URL
+import java.util.*
+import java.util.concurrent.Executors
+import java.util.concurrent.Semaphore
+
+private val logger = LoggerFactory.getLogger(IntegrationService::class.java)
+private const val fetchSize = 10000
+private const val readRateLimit = 1000
+private const val uploadBatchSize = 10000
```

Comment:
super minor but you can insert `_` arbitrarily in numbers so pref `1_000`, `10_000`, `100_000` etc. for faster readability